### PR TITLE
Add EulerPhi and specialize Integer comparisons

### PR DESCRIPTION
**Pull request by rocky**
Every now and then I use Mathics as a hobby item. (That's why I was interested in it in the first place.)

Something I wanted in playing around with Mathics.

<a href="https://gitpod.io/#https://github.com/Mathics3/mathics-core/pull/286"></a>

### CODE REVIEW

File: mathics/builtin/numbers/numbertheory.py [hunk 1]
```diff
@@ -141,11 +143,56 @@ def apply(self, n, evaluation):
...
+        = {1, 1, 2, 2, 4, 2, 6, 4, 6, 4, 10, 4, 12, 6, 8, 8, 16}
+    """
+
+    attributes = listable | numeric_function | protected
+    summary_text = "Euler totient function"
+    sympy_name = "totient"
+
+    def apply(self, n, evaluation):
+        "EulerPhi[n_Integer]"
+        if n < Integer0:
```
Comment (rocky) [inline mathics/builtin/numbers/numbertheory.py:186]: Notice a recent change in the Integer class allows us now to compare two Mathics Integers using "<" and in a more efficient way that was done previously.

File: mathics/builtin/numbers/numbertheory.py [hunk 2]
```diff
@@ -141,11 +143,54 @@ def apply(self, n, evaluation):
...
+        = {1, 1, 2, 2, 4, 2, 6, 4, 6, 4, 10, 4, 12, 6, 8, 8, 16}
+    """
+
+    attributes = listable | numeric_function | protected
+    summary_text = "Euler totient function"
+    sympy_name = "totient"
+
+    def apply(self, n, evaluation):
+        "EulerPhi[n_Integer]"
+        return super().apply(abs(n), evaluation)
```
Comment (rocky) [inline mathics/builtin/numbers/numbertheory.py:186]: Note that the addition of `abs()` in `Integer` of core.py which builds on a recent change simplifies this.

File: mathics/builtin/numbers/numbertheory.py [hunk 3]
```diff
@@ -90,15 +92,15 @@ class Divisors(Builtin):
...
     attributes = listable | protected
+    summary_text = "integer divisors"
 
     def apply(self, n, evaluation):
         "Divisors[n_Integer]"
         if n == Integer0:
             return None
         return Expression(
-            "List", *[from_sympy(i) for i in sympy.divisors(n.to_sympy())]
+            SymbolList, *sympy.divisors(n.to_sympy()), element_conversion_fn=from_sympy
```
Comment (rocky) [inline mathics/builtin/numbers/numbertheory.py:103]: The above removal of a list comprehension was made possible by the addition of `element_conversion_fn` recently. Personally, I think the above is clearer as well.

File: mathics/builtin/numbers/numbertheory.py [hunk 4]
```diff
@@ -171,7 +216,8 @@ def apply(self, n, evaluation):
             factors = sympy.factorint(n.value)
             factors = sorted(factors.items())
             return Expression(
-                "List", *(Expression("List", factor, exp) for factor, exp in factors)
+                SymbolList,
+                *(Expression(SymbolList, factor, exp) for factor, exp in factors)
```
Comment (rocky) [inline mathics/builtin/numbers/numbertheory.py:220]: element_conversion_fn probably could be used with a lambda closure here as well. 

I'll leave this for some other time though.

File: mathics/builtin/numbers/numbertheory.py [hunk 5]
```diff
@@ -182,7 +228,8 @@ def apply(self, n, evaluation):
                 factors[factor] = factors.get(factor, 0) - exp
             factors = sorted(factors.items())
             return Expression(
-                "List", *(Expression("List", factor, exp) for factor, exp in factors)
+                SymbolList,
+                *(Expression(SymbolList, factor, exp) for factor, exp in factors)
```
Comment (rocky) [inline mathics/builtin/numbers/numbertheory.py:232]: Same as above.

**Review by mmatera — APPROVED** [review]
LGTM

**Pull request merged**